From 1fdc99e834fcbb0bf393b92337de2b1d63974163 Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Wed, 25 Sep 2024 17:39:56 +0200
Subject: [PATCH] Actually mark internal helper symbols local

The version script is used but it (by accident?) dropped making
other symbols local, exposing the entirety of libbasic.a into
the final lib.

While at it, we have to link against the internal static copy
of libsystemd because basuctl relies on internal symbols.
---
 meson.build                   | 2 +-
 src/libsystemd/libsystemd.sym | 2 ++
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 6f79707..831f1d3 100644
--- a/meson.build
+++ b/meson.build
@@ -328,7 +328,7 @@ exe = executable('basuctl',
                  'src/busctl/busctl-introspect.h',
                  include_directories : includes,
                  link_with : [libbasic,
-                              libsystemd],
+                              libsystemd_static],
                  install : true)
 
 meson.override_dependency('basu', declare_dependency(
diff --git a/src/libsystemd/libsystemd.sym b/src/libsystemd/libsystemd.sym
index cdb775a..ea4284e 100644
--- a/src/libsystemd/libsystemd.sym
+++ b/src/libsystemd/libsystemd.sym
@@ -324,4 +324,6 @@ global:
 
         sd_bus_set_close_on_exit;
         sd_bus_get_close_on_exit;
+local:
+        *;
 };
-- 
2.46.1

